### PR TITLE
Release MongoDB Kubernetes Operator v0.12.0

### PR DESCRIPTION
**Pull request by MaciejKaras**
### All Submissions:

* [ ] Have you opened an Issue before filing this PR?
* [ ] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

File: docs/how-to-release.md [hunk 1]
```diff
@@ -1,16 +1,24 @@
...
-      * `cd helm-charts && git pull origin main`.
+    * Pull the changes in the helm-charts submodule folder to get the latest main.
+      * `cd helm-charts`
+      * `git submodule update --init` - if submodule was not initialised before
+      * `git pull origin main`
     * Update any changing versions in [release.json](../release.json).
+      * `operator` - always when doing a release 
+      * `version-upgrade-hook`, `readiness-probe` - whenever we make changes in the [versionhook](../cmd/versionhook) or [readiness](../cmd/readiness) files
+      * `agent` - bump to the newest version available
+      * `agent-tools-version` - bump to the newest version available :warning: how to get the newest version?
```
Comment (MaciejKaras) [inline docs/how-to-release.md:12]: > bump to the newest version available :warning: how to get the newest version?
@mircea-cosbuc could help answer this one?

File: docs/how-to-release.md [hunk 2]
```diff
@@ -1,16 +1,24 @@
...
     * Ensure that [the release notes](./RELEASE_NOTES.md) are up to date for this release.
+      * **:warning: how should we know what should be included?**
```
[the remark this answers is not in the document]
Comment (mircea-cosbuc) [reply docs/how-to-release.md:14]: The common process is indeed checking that all merged PRs have a covered entry in the release notes.

File: docs/how-to-release.md (hunk 1 above)
Comment (mircea-cosbuc) [reply docs/how-to-release.md:12]: The latest `agent-tools-version` is coming from the release branch of the agent we're publishing.

**Review by mircea-cosbuc — APPROVED** [review]
LGTM!

**Pull request merged**